app/src/ui/merge-branch/merge.tsx

```diff
@@ -103,6 +104,19 @@ export class Merge extends React.Component<IMergeProps, IMergeState> {
     this.setState({ filterText })
   }
 
+  private onItemClick = (branch: Branch, source: ClickSource) => {
+    if (source.kind !== 'keyboard' || source.event.key !== 'Enter') {
+      return
+    }
+
+    source.event.preventDefault()
+    const { selectedBranch } = this.state
+
+    if (selectedBranch !== null && selectedBranch.name === branch.name) {
```

Comment:
Is a race condition possible here where the clicked branch is different that the selected branch or this is a defensive check to make sure that we don't mess up?

Reply:
The latter, perhaps too defensive since the only way to trigger a _keyboard_ event for a list row is to first move selection to it. For click events there's a chance of a race condition though.